### PR TITLE
[CORDA-1113] Close classloader

### PR DESCRIPTION
**Pull request by m4ksio**
Fixes CORDA-1113, cherry picked from https://github.com/corda/corda/pull/2650

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/ClassloaderUtils.kt [hunk 1]
```diff
@@ -26,6 +26,7 @@ fun scanJarForContracts(cordappJarPath: String): List<ContractClassName> {
     // Only keep instantiable contracts
     val classLoader = URLClassLoader(arrayOf(File(cordappJarPath).toURL()), currentClassLoader)
     val concreteContracts = contracts.map(classLoader::loadClass).filter { !it.isInterface && !Modifier.isAbstract(it.modifiers) }
+    classLoader.close()
```
Comment (tudor-malene) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/ClassloaderUtils.kt]: You can do :
```
URLClassLoader(arrayOf(File(cordappJarPath).toURL()), currentClassLoader).use {
  ..
}
```
it seems URLClassLoader implements Closable

Comment (m4ksio) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/ClassloaderUtils.kt]: Yeah, but this a) would make even bigger code-moster 2) its just a cherry-pick from commit already merged to master :)

Comment (tudor-malene) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/ClassloaderUtils.kt]: but on the other hand, it would close it properly in a "finally" block

Comment (m4ksio) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/ClassloaderUtils.kt]: And that's how you convinced me to change :)

**Pull request merged**